app/controllers/tags_controller.rb

```diff
@@ -94,6 +94,26 @@ def select_merge; end
 
   def merge
     @primary = @tag
+
+    # No merge to self.
+    if params[:merge_with_id].eql? @primary.id.to_s
+      flash[:danger] = 'Cannot merge a tag with itself.'
+
+      # redirect_to tag_path(id: @category.id, tag_id: @primary.id) && return
+
+      # render :edit, status: 400
+      # return
+
+      # render json: { success: false, status: 400, tag: @tag }
+      # return
+
+      # render 'tag', category: @category, tag: @tag
+      # return
+
+      redirect_back fallback_location: categories_path
```

Comment:
This didn't what I was expecting it to do either, at all, but I think the results were unexpectedly great.